File: Example/JSONObjectMapper/Comment.m
//
//  Comment.m
//  JSONObjectMapperExample
//
//  Created by Sergey on 20/05/2018.
//  Copyright © 2018 Sergey. All rights reserved.
//

#import "Comment.h"
#import <JSONObjectMapper/JSONObjectMapping.h>
#import <JSONObjectMapper/JSONRelationshipMapping.h>

@implementation Comment

@dynamic commentID;
@dynamic message;
@dynamic date;
@dynamic user;
@dynamic post;

+ (JSONObjectMapping *)defaultMapping {
    return [JSONObjectMapping mappingWithDictionary:@{
        @"commentID": @"comment_id",
        @"message": @"message",
        @"date": @"date",
        @"user": [JSONRelationshipMapping mappingWithJSONKey:@"user_id"]
    }];
}

+ (NSString *)primaryKey {
    return @"commentID";
}

@end

File: ExampleSwift/ExampleSwift/Comment.swift
//
//  Comment.swift
//  JSONObjectMapper_Example
//
//  Created by Sergey on 21/08/2018.
//

import CoreData
import UIKit

@objc(Comment)
public class Comment: NSManagedObject, JSONMappingProtocol {

    @NSManaged var commentID: NSNumber
    @NSManaged var message: String?
    @NSManaged var date: Date?
    @NSManaged var user: User?
    @NSManaged var post: Post?
    
    public class func defaultMapping() -> JSONObjectMapping! {
        return JSONObjectMapping(dictionary: [
            "commentID": "comment_id",
            "message": "message",
            "date": "date",
            "user": JSONRelationshipMapping(jsonKey: "user_id")
        ])
    }
    
    public class func primaryKey() -> String! {
        return "commentID"
    }
}
